extension/observer/ecsobserver/exporter.go

```diff
@@ -90,29 +89,29 @@ func (e *taskExporter) exportTask(task *taskAnnotated) ([]prometheusECSTarget, e
 
 	// Base for all the containers in this task, most attributes are same.
 	baseTarget := prometheusECSTarget{
-		Source:                 aws.StringValue(task.Task.TaskArn),
+		Source:                 *task.Task.TaskArn,
```

Comment:
Can we really replace all the `aws.StringValue` usages like this? Is it guaranteed to be non-nil?

Reply:
Ah good point, I'm not super familiar with this component, but it doesn't seem like it https://docs.aws.amazon.com/AmazonECS/latest/APIReference/API_ListTasks.html

TaskArn will be populated for sure, but most of the other fields are populated based on the request and are optional fields. I'll swap to `aws.ToString` which maintains the same behavior of defaulting to the zero-value if nil.

Reply:
I've reverted all non-test code changes to the equivalent aws-sdk-go-v2 dereferencing with default function instead of reasoning about which ones are safe to dereference in the latest push - thanks for catching this!